src/cmap/establish/handshake.rs

```diff
@@ -335,6 +335,10 @@ pub(crate) struct Handshaker {
     http_client: crate::runtime::HttpClient,
 }
 
+#[cfg(test)]
+#[allow(clippy::incompatible_msrv)]
```

Comment:
This should be fine since we only test on the latest release and `cfg(test)` isn't transitive, so someone using <1.70 (when `OnceLock` was introduced) can still use the `mongodb` crate fine, including in their tests.